Source/missiles.cpp

```diff
@@ -2843,53 +2835,39 @@ void MI_LArrow(Missile &missile)
 		ChangeLight(missile._mlid, missile.position.tile, missile._miAnimFrame + 5);
 		missile_resistance rst = MissilesData[missile._mitype].mResist;
 		if (missile._mitype == MIS_LARROW) {
-			int mind;
-			int maxd;
-			if (p != -1) {
-				mind = Players[p]._pILMinDam;
-				maxd = Players[p]._pILMaxDam;
-			} else {
-				mind = GenerateRnd(10) + 1 + currlevel;
-				maxd = GenerateRnd(10) + 1 + currlevel * 2;
-			}
+			const auto damage = p != -1
+			    ? Players[p]._pILDamage
+			    : Damage { GenerateRnd(10) + 1 + currlevel, GenerateRnd(10) + 1 + currlevel * 2 };
+
 			MissilesData[MIS_LARROW].mResist = MISR_LIGHTNING;
-			CheckMissileCol(missile, mind, maxd, false, missile.position.tile, true);
+			CheckMissileCol(missile, damage, false, missile.position.tile, true);
 		}
 		if (missile._mitype == MIS_FARROW) {
-			int mind;
-			int maxd;
-			if (p != -1) {
-				mind = Players[p]._pIFMinDam;
-				maxd = Players[p]._pIFMaxDam;
-			} else {
-				mind = GenerateRnd(10) + 1 + currlevel;
-				maxd = GenerateRnd(10) + 1 + currlevel * 2;
-			}
+			const auto damage = p != -1
+			    ? Players[p]._pIFDamage
+			    : Damage { GenerateRnd(10) + 1 + currlevel, GenerateRnd(10) + 1 + currlevel * 2 };
+
 			MissilesData[MIS_FARROW].mResist = MISR_FIRE;
-			CheckMissileCol(missile, mind, maxd, false, missile.position.tile, true);
+			CheckMissileCol(missile, damage, false, missile.position.tile, true);
 		}
 		MissilesData[missile._mitype].mResist = rst;
 	} else {
 		missile._midist++;
 
-		int mind;
-		int maxd;
+		Damage damage { 0 };

```

Comment:
This should be able to use the default constructor, explicit initialisation here is unnecessary since it will always be overwritten